lib/grizzly/zipgateway/exit_monitor.ex

```diff
@@ -0,0 +1,22 @@
+defmodule Grizzly.ZIPGateway.ExitMonitor do
+  @moduledoc false
+
+  use GenServer
+
+  def start_link(_ \\ []) do
+    GenServer.start_link(__MODULE__, [])
+  end
+
+  @impl GenServer
+  def init(_) do
+    Process.flag(:trap_exit, true)
```

Comment:
I don't quite understand this? It doesn't seem like there is any other process to monitor within this one. There also is no `handle_info` to handle any exit signal it does end up receiving?

Reply:
You need to trap exits in order for the terminate callback to be called when receiving an exit signal from the parent process. https://hexdocs.pm/elixir/GenServer.html#c:terminate/2

Reply:
Can the `DynamicSupervisor` that holds the Connections be deleted? From browsing the code, it looks like pids are being used to refer to connections and it seems like they should be linked to a process. At least, it would be nice to get rid the DynamicSupervisor and link them somewhere that they could be cleaned up automatically.

Reply:
>  From browsing the code, it looks like pids are being used to refer to connections

🤔 Connections _should_ be referenced through a registry. If another process is holding on to a Connection's pid, that seems like it could be a source of bugs.

> At least, it would be nice to get rid the DynamicSupervisor and link them somewhere that they could be cleaned up automatically.

I hear you, but I'm not sure what other process we'd want to link them to. In order to keep virtual node ids as stable as possible, Connections need to be a) shared between callers and b) as long-lived as possible.

Reply:
Ok, next idea: Can the Connection registry be a child of `Grizzly.Supervisor`? Basically replace where you have `ExitMonitor` in the child list.

Reply:
Moving the Connection registry and/or dynamic supervisor to be a child of `Grizzly.ZIPGateway.Supervisor` (I assume that's what you meant) would break the ability for Grizzly to run without managing the Z/IP Gateway process. That's not a major issue in embedded environments, but it would (a) break my local dev workflow (running Z/IP Gateway in a VM and Grizzly locally) and (b) require additional restructuring to avoid breaking tests.